common/display/displayplanemanager.cpp

```diff
@@ -111,6 +111,9 @@ std::tuple<bool, DisplayPlaneStateList> DisplayPlaneManager::ValidateLayers(
   auto layer_begin = layers.begin();
   auto layer_end = layers.end();
   bool render_layers = false;
+  // Below flag takes care of condition,when the cursor layer is assigned the
+  // lower zorder
+  bool cursor_plane_enable = true;
```

Comment:
hmm we shouldn't assign cursor plane to primary for any reason.  Is it so that in this case we just have cursor layer and nothing else ? If we have more than one layer perhaps the bug is in present call in drmhwc2 where t layers are expected to be populated in right z order. Can you check whats happening in that case and why we have cursor layer as  first layer?